src/components/StatusesList/index.tsx

```diff
@@ -1,19 +1,67 @@
+import { STATES } from 'types/SigningStates'
 import { StatusText } from 'components/Text'
+import AppStore from 'stores/AppStore'
+import Button from 'components/Button'
 import ChildrenProp from 'models/ChildrenProp'
 import Delimiter from 'components/Delimiter'
+import JobStore from 'stores/JobStore'
 import StatusCard from 'components/StatusCard'
+import classnames, { display, gap } from 'classnames/tailwind'
+import generateCommitment from 'helpers/generateCommitment'
+import startGeneration from 'helpers/proofs/startGeneration'
+
+const buttonsWrapper = classnames(display('flex'), gap('gap-x-4'))
 
 export default function ({
   statusDescription,
+  hasError,
   children,
 }: ChildrenProp & {
-  statusDescription: string
+  hasError?: boolean
+  statusDescription?: string
 }) {
   return (
     <StatusCard>
       {children}
-      <Delimiter horizontal />
-      <StatusText>{statusDescription}</StatusText>
+      {statusDescription ? (
+        <>
+          <Delimiter horizontal />
+          <StatusText>{statusDescription}</StatusText>
+        </>
+      ) : undefined}
+      {hasError ? (
+        <div className={buttonsWrapper}>
+          <Button
+            small
+            onClick={async () => {
+              if (
+                AppStore.flowState === STATES.GENERATE_COMMITMENT &&
+                AppStore.proof
+              ) {
+                await generateCommitment(AppStore.proof)
+                return
+              }
+
+              await startGeneration({
+                generationWay: AppStore.preferredProofWay,
+                proverAddress: JobStore.proverAddress,
+              })
+            }}
+          >
```

Comment:
Let's move it outside JSX, wdyt? I think it's better to create a separate function and pass it into `onClick` method